pkg/pgmodel/ingestor/copier.go

```diff
@@ -174,30 +174,51 @@ func copierGetBatch(ctx context.Context, batch []readRequest, in <-chan readRequ
 
 	req, ok := <-in
 	if !ok {
-		return batch, false
+		return batch, stats, false
 	}
 	span.AddEvent("Appending first batch")
-	batch = append(batch, req)
+	copyRequest, ok := <-req.copySender
+	if !ok {
+		return batch, stats, false
+	}
+	batch = append(batch, copyRequest)
+	stats.Update(copyRequest)
+
+	totalPoints := atomic.LoadInt64(&metrics.BatchingPoints)
+	//try to get a fair number of points across copiers
+	targetPoints := int(totalPoints) / numCopiers
```

Comment:
The key change is this line which specifies how many points a batch should be instead of relying on a constant number of metrics (100 as was the case previously)